invoiceregistry.go

```diff
@@ -895,6 +915,43 @@ func (i *InvoiceRegistry) requestSettle(ctx context.Context,
 
 	hash := state.invoice.PaymentPreimage.Hash()
 
+	// Delete in-memory record for this invoice. Only open invoices are
+	// kept in memory.
+	delete(i.invoices, hash)
+
+	// Notify subscribers that the htlcs should be settled
+	// with our peer.
+	for key := range state.acceptedHtlcs {
+		htlcSettleResolution := NewSettleResolution(
+			state.invoice.PaymentPreimage, ResultSettled,
+		)
+		i.notifyHodlSubscribers(key, htlcSettleResolution)
+	}
+
+	// TODO: Wait for final settle event from lnd. Unfortunately this
+	// event is not yet implemented.
+
+	// Mark invoice as settled.
+	if err := i.cdb.Settle(ctx, hash); err != nil {
+		return errors.New("cannot settle invoice in database")
```

Comment:
would it make sense to wrap the original error here to get more insight about what's failing?

Reply:
yes, done